casper/src/main/scala/coop/rchain/casper/api/BlockAPI.scala

```diff
@@ -44,19 +52,70 @@ object BlockAPI {
         for {
           status <- casper.addBlock(b)
         } yield addResponse(status.some, b.some),
-      DeployServiceResponse(false, "Error: Casper instance not available")
+      DeployServiceResponse(success = false, "Error: Casper instance not available")
     )
 
   def createBlock[F[_]: Monad: MultiParentCasperConstructor: Log]: F[DeployServiceResponse] =
     MultiParentCasperConstructor.withCasper[F, DeployServiceResponse](
       casper =>
         for {
           maybeBlock <- casper.createBlock
-          status     <- maybeBlock.traverse(casper.addBlock(_))
+          status     <- maybeBlock.traverse(casper.addBlock)
         } yield addResponse(status, maybeBlock),
-      DeployServiceResponse(false, "Error: Casper instance not available")
+      DeployServiceResponse(success = false, "Error: Casper instance not available")
     )
 
+  def getListeningNameResponse[
+      F[_]: Monad: MultiParentCasperConstructor: Log: SafetyOracle: BlockStore](
+      listeningName: Channel): F[ListeningNameResponse] = {
+    def casperResponse(implicit casper: MultiParentCasper[F], channelCodec: Codec[Channel]) =
+      for {
+        estimates           <- casper.estimator
+        tip                 = estimates.head
+        internalMap         <- BlockStore[F].asMap()
+        mainChain           = ProtoUtil.getMainChain(internalMap, tip, IndexedSeq.empty[BlockMessage])
+        maybeRuntimeManager <- casper.getRuntimeManager
+        runtimeManager      = maybeRuntimeManager.get // This is safe. Please reluctantly accept until runtimeManager is no longer exposed.
+        maybeBlocksWithActiveName <- mainChain.toList.traverse { block =>
+                                      val serializedLog =
+                                        block.body.fold(Seq.empty[Event])(_.commReductions)
+                                      val log =
+                                        serializedLog.map(EventConverter.toRspaceEvent).toList
+                                      val listeningNameReduced = log.exists {
+                                        case Produce(channelHash, _) =>
+                                          channelHash == StableHashProvider.hash(
+                                            immutable.Seq(listeningName))
+                                        case Consume(channelHash, _) =>
+                                          channelHash == StableHashProvider.hash(
+                                            immutable.Seq(listeningName))
```

Comment:
Per @guardbotmk3 We should sort the term before we go searching for it in the tuplespace since there is no guarantee that the proto someone gives us is already sorted.